File: tests/rosetta/transpiler/php/babbage-problem.php
<?php
ini_set('memory_limit', '-1');
$target = 269696;
$modulus = 1000000;
$n = 1;
while (true) {
  $square = $n * $n;
  $ending = $square % $modulus;
  if ($ending == $target) {
  echo rtrim('The smallest number whose square ends with ' . json_encode($target, 1344) . ' is ' . json_encode($n, 1344)), PHP_EOL;
  break;
}
  $n = $n + 1;
}

File: tests/rosetta/transpiler/Kotlin/babbage-problem.kt
import java.math.BigInteger

val target: Int = 269696
val modulus: Int = 1000000
var n: Int = 1
fun main() {
    while (true) {
        val square: Int = n * n
        val ending: Int = Math.floorMod(square, modulus)
        if (ending == target) {
            println((("The smallest number whose square ends with " + target.toString()) + " is ") + n.toString())
            break
        }
        n = n + 1
    }
}
